python/packages/nisar/products/writers/BaseL2WriterSingleInput.py

```diff
@@ -1996,6 +2041,105 @@ def geocode_metadata_group(self,
         input_raster_obj = isce3.io.Raster(
             input_temp.name, raster_list=input_raster_list)
 
+        geocode_kwargs = {}
+        if (lines == 1 or samples == 1):
+            geocode_kwargs['data_interpolator'] = 'nearest'
+
+        elif (lines < 5 or samples < 5):
+            geocode_kwargs['data_interpolator'] = 'bilinear'
+
+        # If geocoding the noise-equivalent backscatter LUT for GCOV products,
+        # the terrain radiometry convention needs to be updated from
+        # beta0/sigma0 to gamma0
+        flag_apply_rtc = (flag_noise_equivalent_backscatter and
+                          self.product_type == 'GCOV')
+
+        geocode_kwargs['flag_apply_rtc'] = flag_apply_rtc
+
+        if flag_apply_rtc:
+            geocode_kwargs['input_terrain_radiometry'] = \
+                self.cfg['processing']['rtc']['input_terrain_radiometry_enum']
+            geocode_kwargs['output_terrain_radiometry'] = \
+                self.cfg['processing']['rtc']['output_type_enum']
+
+        self.geocode_raster(input_raster_obj,
+                            output_h5_group_path,
+                            output_ds_name_list,
+                            radar_grid,
+                            metadata_group,
+                            compute_stats,
+                            **geocode_kwargs)
+
+        input_temp.close()
+
+        return flag_all_succeeded
+
+    def geocode_raster(self,
+                       input_raster_obj,
+                       output_h5_group_path,
+                       output_ds_name_list,
+                       radar_grid,
+                       metadata_group,
+                       compute_stats,
+                       data_interpolator=None,
+                       **geocode_kwargs):
+        """
+        Geocode an ISCE3 Raster object containing look-up tables (LUTs)
+        radar coordinates to the output product in map coordinates
+        using runconfig parameters associated with that
+        metadata group, either 'calibrationInformation'
+        or 'processingInformation'
+
+        Parameters
+        ----------
+        input_raster_obj: isce3.io.Raster
+            Raster object to geocode
+        output_h5_group_path: str
+            Path of the output group
+        output_ds_name_list: str, list
+            List of output LUT datasets
+        radar_grid: isce3.product.RadarGridParameters
+            RadarGridParameters object representing the geometry of the
+            input raster object
+        metadata_group: str
+            Metadata group, either 'calibrationInformation'
+            or 'processingInformation'
```

Comment:
```suggestion
            or 'processingInformation'.
            Used to determine which geo grid to geocode to.
```

Reply:
Consider passing the geo grid object itself here instead of passing the `metadata_group` name.